extension/public/reddit.js

```diff
@@ -5,95 +5,117 @@ var Reddit = {
-  },
+  //counter for checkForUpdate
+  counter: 0,
+  refreshFrequency: 100,
 
-  // HTML for dashboard
-  dashboardHTML: function () {
-    var h = '<div class="reddit-dashboard">';
-    h += '<div class="reddit-page-status">';
-    h +=
-      '<div class="reddit-status total_hidden"><span class="variable"></span>Total hidden</div>';
-    h += "</div>";
-    h += '<div class="reddit-controls">';
-    h += '<div class="reddit-control hide_links">Load Hide Buttons</div>';
-    h += '<div class="reddit-control unhide_links">Unhide all</div>';
-    h += "</div>";
-    h += "</div>";
-
-    return h;
+  // initalize
+  initialize: function () {
+      this.blurImage();
+      this.checkForUpdate();
   },
 
-  // initalize UI
-  initialize: function () {
-    this.setupUI();
+  //listener to scroll events
+  //updates new visible posts to be filtered
+  checkForUpdate: function(){
+      document.addEventListener('scroll', (ev) => {
+      this.counter++;
+      if(this.counter == this.refreshFrequency){
+          this.counter = 0;
+          this.blurImage();
+          console.log("updated");
+      }
+      });
   },
 
-  // initalize all UI elements and onclick binding
-  setupUI: function () {
-    list = document.querySelector(this.listSelector);
-    list.innerHTML = this.dashboardHTML() + list.innerHTML;
 
-    document
-      .querySelector(".reddit-control.unhide_links")
-      .addEventListener("click", () => {
-        this.restoreDOM();
-        console.log("unhide all");
-      });
 
-    document
-      .querySelector(".reddit-control.hide_links")
-      .addEventListener("click", () => {
-        this.setupPostUI();
-        console.log("all links hidden");
-      });
+  //check if any configured keywords are found in a post
+  //return true if found
+  async filterText (post,callBack) {
+      //temporary filter data
+      //these data must be stored in cookie
+      const blockedWordsData = [
+          "denr",
+          "nobody",
+          "taranaki",
+          "covid",
+          "isis",
+          "pizza"
+      ];
+
+      const title = this.getDOMTitle(post);
+      const description = this.getDOMDescription(post);
+      //image varible contain image url then contain result of ocr
+      let image = this.getDOMImageLink(post);
+
+      if(image != "n/a"){
+          image = await this.ocr(image);
+          image = image.toLowerCase();
+          //callback to wait ocr to be processed
+          callBack();
+      }
+
+      console.log(image);
+      //search if blockedwords are in the title
+      //true if found
+      for (i in blockedWordsData){
+          console.log("pass");
+          if(title.search(blockedWordsData[i]) != -1){
+            return true;
+          }else if (image.search(blockedWordsData[i]) != -1){
+            return true;
+          }else if (description.search(blockedWordsData[i]) != -1){
+            return true;
+          }
+      }
```

Comment:
`.search` doesn't really make sense here, we should be using `.contains`. Also, having an expression which then returns a condition is generally bad practice.

Reply:
agreed, I should change it